### PR TITLE
Ignore dotnet-test-xunit exit code if KOREBUILD_IGNORE_DOTNET_TEST_EXIT_CODE is set

### LINKED ISSUE #78: Ignore non-zero exit from dotnet-test-xunit when running under TeamCity
The xunit runner is supposed to return a 0-exit code when one or more tests fail when running under TeamCity. TeamCity will report these errors and fail the build. However it returns a non-zero exit code. This prevents features such as test muting from working. 

The fix here would be to ignore the exit code in `_dotnet-test.shade` when running in TeamCity.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #78

cc @cesarbs \ @troydai

### CODE REVIEW

File: build/shade/_dotnet-test.shade [hunk 1]
```diff
@@ -37,7 +38,18 @@ default configuration = 'Debug'
         }
 
         testArgs += noParallelTestProjects.Contains(projectName) ? " -parallel none" : "";
-        Dotnet("test" + testArgs, projectFolder);
+
+        try
+        {
+            Dotnet("test" + testArgs, projectFolder);
+        }
+        catch
```
Comment (troydai) [inline build/shade/_dotnet-test.shade:46]: could this accidentally swallow any exception which really should have failed the test?

Comment (pranavkm) [inline build/shade/_dotnet-test.shade:46]: There's a risk of it, but there's not much we could do short of matching the exception message here: https://github.com/sakeproject/sake/blob/master/src/Sake.Library/Shared/_exec.shade#L60

**Comment by natemcmaster** [comment]
Why does dotnet-test-xunit return non-zero?

**Comment by pranavkm** [comment]
@natemcmaster it seems like a bug in the runner. This was stop gap to allow us to mute tests until this is fixed.

**Comment by natemcmaster** [comment]
Is there a bug on dotnet-test-xunit? Under what conditions do tests pass and it still returns a non-zero exit code?

**Comment by pranavkm** [comment]
@natemcmaster it's the other way around. The tests fail and TeamCity is aware of it, but the test runner fails with a non-zero exit code which causes the build to fail. Muting doesn't help in this case.

**Comment by natemcmaster** [comment]
Thanks for clarifying. Could we key this behavior off the "IfTeamCity" variable instead?

**Comment by pranavkm** [comment]
I didn't want to turn it on the CoreCLR build in case there was something that broke the runner.

**Pull request closed without merging**